### ISSUE TITLE
Compilation issue Kernel 5.4 

### ISSUE THREAD

**Issue by Bouliere83** [oracle/ktf#18]
Hi,

I was using KTF on a project with a Linux Kernel 5.3 (Ubuntu 18.04), but after passing on 5.4 (Ubuntu 20.04), the KTF Framework cannot compile.
From what i saw it is due to this patch :
[https://github.com/torvalds/linux/commit/9adeaa226988b97bc15928e12f40a9863134467c](url) 
The memcg_cache_params is not define anymore in a public header, instead is it located in a internal header.
Do you have any plans to fix it?

Thanks you.

**Comment by Yao-J** [comment]
Yes, i met same error on Kernel 5.4 , so is it expected when can fix it ?

**Comment by alan-maguire** [comment]
The fix is simple but needs to be pushed; I'm working on getting that done. Hopefully should be sometime over the next few days.

**Comment by Bouliere83** [comment]
Ok that some good news, thank you.

**Comment by alan-maguire** [comment]
finally got write access; I've pushed a bunch of changes that Knut and I authored that should hopefully resolve various compilation issues; let me know if you run into any trouble. thanks!

**Comment by Yao-J** [comment]
Hi, alan-maguire 
Could you show which commit can fix this issue ？ 
Actually, this issue is related to struct memcg_cache_params, but i can't find related fix in the commit log. The latest fix is "ktf: replace struct timespec with struct timespec64"

**Comment by knuto** [comment]
Take a look at [this commit](https://github.com/oracle/ktf/commit/656da8cf98ef7e245fe930bd89c8eb3557324ca9)

**Comment by Yao-J** [comment]
Hi, 
new commit can fix this issue, but it will prompt "error: unknown warning option '-Wno-packed-bitfield-compat'" in lib/Makefile
This option available on kernel 4.14 , wonder why it can't pass on 5.4 ?

**Comment by knuto** [comment]
Are you perhaps not using gcc? The option seems present still? Anyway you can safely just remove the option from Makefile.am as KTF does not use bitfields, this is just a remains from the predecessor of KTF and should have been removed.
